### ISSUE TITLE
command 'hl7tools.DisplaySegmentFields' not found

### ISSUE THREAD

**Issue by Rod-in-NM** [RobHolme/vscode-hl7tools#3]
I just downloaded the HL7Tools for VS Code. I've got a HL7 file, renamed the extension to .hl7, then opened it with Code. It identified it as an HL7 file. I'm looking over your comments for version 1.6.2. I thought it would be good to try various things, like hovering over a segment to get the field name. That didn't work. So then I saw something about displaying segment fields. So, I selected some segments, pressed F1 then entered HL7 Tools: Display Segment Fields. I got the following error:

> command 'hl7tools.DisplaySegmentFields' not found

I'm wondering if I didn't download something that's required? Or if its some other problem?

I'm working with VS Code version 1.15.1

**Comment by RobHolme** [comment]
Sounds like an issue with the extension. The hover descriptions should be applied automatically when the file is opened. The command not found error generally indicates the command has encountered an exception/error. 

Do you have a sample hl7 file that you have the issue with that you can share? Could you also let me know what platform you are running (Windows, Linux, Mac). This should help me to reproduce the issue.

Alternatively, if you can't share the file that triggers the issue, could you run the extension in debug mode and send me a copy of the debug console - this might help pinpoint the issue. To do this:
1) Clone or download this repository (https://github.com/RobHolme/vscode-hl7tools), saving the folder to your workstation. 
2) Open the folder in Visual Studio Code.
3) Press f5 - this will launch a second VS Code window. In this instance of VS Code open the hl7 file that triggers the issue.
4) In the first VS Code window, there should be a debug console visible (if not select 'Debug Console' from the view menu). Copy and reply with the contents of the debug console.

**Comment by Rod-in-NM** [comment]
Hi Rob, thank you for your help on this. I've done as you asked and downloaded the repo to my machine. Got into VSC, pressed F5. Opened the .HL7 file. Still nothing happened when I hovered over fields. Here is the contents from the debug console:

'C:\Program Files (x86)\Microsoft VS Code\Code.exe' --debugBrkPluginHost=29997 --debugId=bd650369-ef69-487f-a143-3b747f503f62 --extensionDevelopmentPath=d:\GitHub\vscode-hl7tools-master\vscode-hl7tools-master 

The extension "hl7tools" is now active.
HL7 file extension detected
HL7 version could not be detected. Defaulting to v2.7.1 schema.
Updating field hover descriptions

**Comment by RobHolme** [comment]
Thanks for providing the debug log - unfortunately no errors logged as I was expecting. If it's possible to send me a copy of the file you are testing with it would help me to reproduce the issue.

If you are unable to provide a copy, I've attached a test HL7 file that works for me - can you confirm if the issue occurs when you view this file (to try and determine if the issue relates to your environment, or the specific hl7 files you view). 
 
[test.zip](https://github.com/RobHolme/vscode-hl7tools/files/1254327/test.zip)

**Comment by Rod-in-NM** [comment]
I was able to download your test.zip file. Thank you for providing a link to it. And I’ve tested it. It works perfectly fine with your test.hl7 file. Popups occur as I move my mouse over the fields. Wonderful!

Unfortunately, I really cannot give you a section of the field. I can clearly see some PHI information which would be bad for me to share with you, as that would be a HIPAA violation. However, I can share with you the first line of the files that I’m having to deal with. These are more like a header and they don’t match at all the file you sent me. Here it is:

PatientMPI|PatientLastName|PatientFirstName|PatientMiddleName|PatientDOB|PatientAge|PatientGender|PatientAlias|PatientStreetAddress|PatientCity|PatientState|PatientZip|PatientCounty|PatientHomePhone|PatientWorkPhone|PatientEthnicity|PatientRace|PatientTribalAffiliation|PatientOccupation|PatientIndustry|PatientSSN|PatientMRN|AlternateContactName|AlternateContactPhone|AlternateContactRelationship|ExpectedSourceOfPayment|CreatedDate|ReportDateTime|ReportingFacilityName|ReportingFacilityAddress|ReportingFacilityPhone|ReportingFacilityCLIA|OrderingFacilityName|OrderingFacilityAddress|OrderingFacilityPhone|OrderingProviderName|OrderingProviderAddress|OrderingProviderPhone|OrderingProviderID|AccessionNumber|OrderTestCode|OrderTestName|CollectionDate|CollectionMethod|Specimen|SpecimenSource|RepeatRedrawInd|ReasonForTest|ReportStatus|OrderComment|ResultPerformingFacilityName|ResultPerformingFacilityCLIA|ResultedTestLocalCode|ResultedTestLocalName|ResultedTestLOINCCode|ResultedTestLOINCName|ResultValueLocalCode|ResultValueLocalDesc|ResultValueSnomedCode|ResultValueSnomedDesc|ResultText|ResultUnits|ReferenceRange|TestInterpretation|TestStatus|ResultDateTime|MethodType|ResultComment|TranslatedGender|TranslatedRace|TranslatedEthnicity|TranslatedCounty|TranslatedExpectedSourcePmt


I hope this is helpful.


Rod

**Comment by RobHolme** [comment]
Hi Rod,

Based on the header line you shared, it does not look like you are viewing a HL7 file. It looks more like a pipe delimited text file / csv file. I would usually expect a HL7 file to include a MSH segment as the first line.

The extension is designed specifically for viewing/editing HL7 v2.x formatted files. The hover descriptions (and other commands) expect the message to conform to the HL7 2.x message schemas (http://www.hl7.org/implement/standards/product_brief.cfm?product_id=185). Running the commands against non HL7 formatted files would lead to unpredictable results (in most cases the commands should exit without performing any actions). This extension is of little value for viewing non HL7 files.  

regards,
Rob.

**Comment by Rod-in-NM** [comment]
Oh WOW! That explains a lot!! I am not at all a HL7 expert, so I didn't have a clue that what I sent you and what we consider proper HL7, isn't in fact HL7! Man. I wonder how where I work got onto using whatever it is we use and then require other agencies submitting data to us to comply with our non-conformity? I wonder if we're using some older version of the HL7 standard?

Thank you Rob, for your help with this. I'll keep the extension, even if it doesn't do me any good. Who knows, maybe we'll eventually adopt the HL7 2.x message schemas.

**Comment by cragun47** [comment]
Adding this comment for the next person with this issue.  I saw this when I was viewing an HL7 file.  Previously the plug-in had worked just fine, but stopped working with `command 'hl7tools.DisplaySegmentFields' not found`.

My solution was to restart VS Code.   (I also uninstalled and reinstalled, but things only worked once I restarted VS Code, so I think next time I would try that first.)


Great tool, @RobHolme !
